test/viewmodels/dashboard_viewmodel_test.dart

```diff
@@ -0,0 +1,178 @@
+// FLUTTER / DART / THIRD-PARTIES
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:notredame/core/constants/preferences_flags.dart';
+
+// MANAGERS
+import 'package:notredame/core/managers/settings_manager.dart';
+
+// VIEW-MODEL
+import 'package:notredame/core/viewmodels/dashboard_viewmodel.dart';
+
+import '../helpers.dart';
+
+// MOCKS
+import '../mock/managers/settings_manager_mock.dart';
+
+SettingsManager settingsManager;
+DashboardViewModel viewModel;
+
+void main() {
+  // Needed to support FlutterToast.
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  // Cards
+  final Map<PreferencesFlag, int> dashboard = {
+    PreferencesFlag.aboutUsCard: 0,
+    PreferencesFlag.scheduleCard: 1,
+    PreferencesFlag.progressBarCard: 2
+  };
+
+  // Reorderered Cards
+  final Map<PreferencesFlag, int> reorderedDashboard = {
+    PreferencesFlag.aboutUsCard: 1,
+    PreferencesFlag.scheduleCard: 2,
+    PreferencesFlag.progressBarCard: 0
+  };
+
+  // Reorderered Cards with hidden scheduleCard
+  final Map<PreferencesFlag, int> hiddenCardDashboard = {
+    PreferencesFlag.aboutUsCard: 0,
+    PreferencesFlag.scheduleCard: -1,
+    PreferencesFlag.progressBarCard: 1
+  };
+
+  group("DashboardViewModel - ", () {
+    setUp(() async {
+      // Setting up mocks
+      settingsManager = setupSettingsManagerMock();
+      setupFlutterToastMock();
+
+      viewModel = DashboardViewModel(intl: await setupAppIntl());
+    });
+
+    tearDown(() {
+      unregister<SettingsManager>();
+      tearDownFlutterToastMock();
+    });
+
+    group("futureToRun - ", () {
```

Comment:
You could add a test to see what happens if the settings manager behave correctly

Reply:
Do you mean incorrectly ? Or is there more to test when it behaves correctly ?

Reply:
You right I wanted to say 'incorrectly' 😅